src/guide/advanced/wasm.md

```diff
@@ -16,36 +16,41 @@ the blocks, but is also easy to understand. Something that looks like your
 traditional `if` statements and `for` loops.
 
 However, in the interim, we have decided to use another portable binary
-standard called **Web assembly** or **WASM**. The main advantage of this
-format is that you can use any language you like (as long as it links
-statically against our helper library), and produce a 32-bit portable
-executable. The compilers take care of the optimisation, and you don't have
-to learn a new language (ahem… solidity… ahem), just to operate on the
-blockchain.
+standard called **Web assembly**, or **WASM**.
+
+## Working with WASM
+
+The main advantage of using the WASM format is that you can use any
+language you like (as long as it links statically against our helper
+library), and produce a 32-bit portable executable. The compilers take care
+of the optimisation, and you don't have to learn a new language (ahem...
+solidity... ahem), just to operate on the blockchain.
 
 You'd still need to use ISI from inside your WASM binary to do anything
-on-chain, but you can do other things, and write e.g. Rust instead of
+on-chain, but you can do other things, and write in Rust instead of
 composing Generic `structs` inside a macro to get conditional logic.
```

Comment:
Not sure about the second part of the sentence starting with "but". Maybe I'm missing something but it doesn't connect with the first half of the sentence

Reply:
Added an explanation.